qiskit_optimization/algorithms/grover_optimizer.py

```diff
@@ -313,7 +313,7 @@ def _get_probs(self, qc: QuantumCircuit) -> Dict[str, float]:
             state = result.get_counts(qc)
             shots = self.quantum_instance.run_config.shots
             hist = {key[::-1]: val / shots for key, val in state.items() if val > 0}
-            self._circuit_results = {b[::-1]: np.sqrt(v / shots) for (b, v) in state.items()}
+            self._circuit_results = {b: np.sqrt(v / shots) for (b, v) in state.items()}
```

Comment:
```suggestion
            self._circuit_results = {b[::-1]: np.sqrt(v / shots) for (b, v) in state.items()}
```

Reply:
I accidentally removed this code `self._circuit_results = {i[-n_key:]: v for i on line 247 when I removed the change of #96. That was causing an error. Please revert this change. Line 316 needs to be `self._circuit_results = {b: np.sqrt(v / shots) for (b, v) in state.items()}`

Reply:
I reverted it according to your suggestion.

Reply:
After I reverted it, unit tests of Grover fail. Are you sure `self._circuit_results = {b: np.sqrt(v / shots) for (b, v) in state.items()}` is correct?

Reply:
Yaya, we also need to add `self._circuit_results = {i[-n_key:]: v for i` as I commented in the above though.